design/specs/table-import.md

```diff
@@ -14,153 +14,175 @@ Adding a new table from a file import allows users to input data into Mathesar w
 
-### Buttons
-Buttons allow users to perform actions and obtain a result from the system.
+#### Abandoning the File Processing Step
 
-#### Types
-There are different types of buttons, and their use depends on several factors, e.g., the type of task they are to perform and their context.
+![](https://share.balsamiq.com/c/m6Ui2naMGZXijJgjcaF15f.png)
 
-![](/assets/design/specs/table-import/rkEwqImqd.png =200x)
+Since technically background/parallel imports are difficult, we won't be supporting that in the user journey as well. When the user tries to move away there is a warning modal that warns them that they would lose the progress and they would land to the first screen if they land back to the import. 
 
-Primary buttons are used for actionable (e.g., submit content, apply a change) and some exceptional navigational purposes (e.g., continue to a next step). However, their use should be restricted to priority or practical actions, with the ideal number of primary buttons being one per view.
+### Confirming Imported Data
 
-Secondary buttons are used alone or in combination with primary controls to complement priority actions.
+![](https://share.balsamiq.com/c/ds5qgkUM5bhGBWvkucdWnT.png)
 
+When the import is successful, the user lands on the active confirmation view.
 
-### Dialogs
-Dialogs are a type of modal window that appears in front of content and is used when the system requires users to make decisions.
-When active, dialogs will prevent other content from being interacted with. In most cases, they are driven by user action, and their unprompted use is discouraged.
+#### Go back button
 
-![](/assets/design/specs/table-import/Hk02qLQc_.png =280x)
+This will take them to import screen. So before that there will be a warning modal which is listed in [Behavior section]. If the table gets confirmed it is added to the schema or else its discarded.
 
-### System Notifications
+#### Selected file section
 
-#### Error Messages
-![](/assets/design/specs/table-import/SJBTzvX9d.png)
+This consists of the file the user uploaded. They can access it from here when they click on it. If the reupload button is selected it has a similar behavior like go back button.
 
+#### Table name
 
+The input has a default name that can be changed by user there itself.
+
+#### Import Summary (Total Records, Columns)
+
+This has a small summary like number of rows, records etc. The number of missing data and duplicated columns is present on the confirmation view. Although user can ignore those warnings and table can be confirmed as is.
+
+#### Options
+
+- Use headers - This will use the first row as headers if selected or else there will be default column names. This will remove duplicate column scenario.
+- Set missing data as null - This will set missing data as null and remove missing data scenario.
+
+#### Verifying and Updating Data Types
```

Comment:
This was the main problem that this spec needed to solve, we need more detail on the data type changing menu. 

Also, we're trying to have visual consistency with the table here, but I don't see anything about that in the spec.